### PR TITLE
Several improvements to vSphere cleanup

### PR DESCRIPTION
**Pull request by ljfranklin**
- Speeds up leftovers on vSphere from ~5 minutes on our vCenter env to a
  couple seconds
  - The `finder.Find` call was super slow on our vCenter as it searches
    through all kinds of resources even though we only care about folders
    and VMs at this time, so I switched to
    `searcher.FindByInventoryPath('/DATACENTER/vm/FILTER')` which is way
    faster
- Previously you needed to run leftovers once to delete the VMs then
  again to delete the empty folders. Now both VMs and folders are
  deleted in one pass
- The previous implementation had limitations where VMs were only
  deleted if they appeared in a nested (grandchild) folder but not if
  they appeared right under the parent folder. This works now
- The previous implementation only handled one level of nested folders,
  now it recursively handles as many nesting levels as needed
- Running vsphere acceptance tests now requires new
  BBL_VSPHERE_VCENTER_DS env var to create fixture VMs. This env var is
  not required to actually run leftovers, just test setup

Potential future improvements:
- Running `leftovers -i vsphere -f my-folder` requires 'my-folder' to be
  in the top-level directory, meaning `leftovers -i vsphere -f
  my-nested-folder` returns no matches. Should this be supported?
- Similarly running `leftovers -i vsphere -f my-vm` doesn't work as
  leftovers expects a top-level folder. Should this be supported?

I personally don't need the two future improvements suggestions for my
current use case but figured I'd write them down as an observation.

### CODE REVIEW

**Comment by genevieve** [comment]
This looks awesome! Thank you for adding an acceptance test - vsphere is really tricky and the more acceptance-level testing there is for changes, the better. 

👍

File: acceptance/vsphere.go [hunk 1]
```diff
@@ -34,37 +39,73 @@ func NewVSphereAcceptance() VSphereAcceptance {
...
+func (v *VSphereAcceptance) CreateFolder(root, name string) *object.Folder {
+	rootFolder := v.FindFolder(root)
 
-	vCenterUrl.User = url.UserPassword(v.VCenterUser, v.VCenterPassword)
+	ctx := context.Background()
+	folder, err := rootFolder.CreateFolder(ctx, name)
+	Expect(err).NotTo(HaveOccurred())
 
-	vContext, cancel := context.WithTimeout(context.Background(), time.Minute*5)
-	defer cancel()
```
[the remark this answers is not in the document]
Comment (ljfranklin) [reply acceptance/vsphere.go]: The timeout moved up into the `BeforeEach` where we're initializing the client. In any case the calls now take just a couple seconds on our vSphere env after replacing the super slow `finder.Find` calls.

File: acceptance/vsphere.go [hunk 2]
```diff
@@ -34,37 +39,73 @@ func NewVSphereAcceptance() VSphereAcceptance {
...
 	ctx := context.Background()
+	rootPool, err := finder.ResourcePoolOrDefault(ctx, "")
+	Expect(err).NotTo(HaveOccurred())
 
-	rootFolder, err := finder.Folder(ctx, root)
+	task, err := folder.CreateVM(ctx, *spec, rootPool, nil)
 	Expect(err).NotTo(HaveOccurred())
 
-	_, err = rootFolder.CreateFolder(ctx, name)
+	_, err = task.WaitForResult(ctx, nil)
```
[the remark this answers is not in the document]
Comment (ljfranklin) [reply acceptance/vsphere.go]: good call, didn't know that was a method. Fixed

**Comment by ljfranklin** [comment]
Just force pushed a change to use `task.Wait()`, let me know if you see any other changes.

**Comment by rowanjacobs** [comment]
LGTM!

**Pull request merged**